mev_inspect/classifiers/utils.py

```
@@ -0,0 +1,89 @@
+from typing import Optional, List, Sequence
+
+from mev_inspect.schemas.swaps import Swap
+from mev_inspect.schemas.transfers import Transfer, ETH_TOKEN_ADDRESS
+
+from mev_inspect.schemas.traces import DecodedCallTrace, ClassifiedTrace
+
+
+def create_swap_from_transfers(
+    trace: DecodedCallTrace,
+    recipient_address: str,
+    prior_transfers: List[Transfer],
+    child_transfers: List[Transfer],
+) -> Optional[Swap]:
+    pool_address = trace.to_address
+
+    if recipient_address is None:
+        return None
+
+    transfers_to_pool = []
+
+    if trace.value is not None and trace.value > 0:
+        transfers_to_pool = [_build_eth_transfer(trace)]
+
+    if len(transfers_to_pool) == 0:
+        transfers_to_pool = _filter_transfers(prior_transfers, to_address=pool_address)
+
+    if len(transfers_to_pool) == 0:
+        transfers_to_pool = _filter_transfers(child_transfers, to_address=pool_address)
+
+    if len(transfers_to_pool) == 0:
+        return None
```

These 3 conditionals I think that deserve a comment. It's not clear what's happenineth_sendPrivateTransactiong. First we try to find prior transfers, if it fails, we try to find child transfers, is that right?

This prior transfers thing is new, right? I don't understand how to find them. It would be nice to have a documented guide of how to make a new swap classifier.

And again, I think it would be clearer to raise an exception instead of returning None. Like exception! the trace is not a swap because it doesn't have transfers to the pool.

---

I agree this would make things much clearer. The goal here was to port the existing logic from swaps into the classifiers, no logic changes. I think this would be better addressed in its own PR, then we can throw for swaps, transfers, liquidations, etc, and address this properly as well as maintain this concise. Thoughts?

---

I agree.